libsolutil/CommonIO.cpp

```diff
@@ -78,6 +78,7 @@ string solidity::util::readStandardInput()
 		ret.append(tmp);
 		ret.append("\n");
 	}
+	ret.pop_back();
```

Comment:
If the while condition is not met, won't this `pop` something that it wasn't supposed to?

Reply:
Yes; in my naivety, I thought that was OK. I didn't want to add logic to the loop checking for the last line each iteration; figured it was better to have one function call afterwards to remove the one unwanted character. Of course, I am eager to hear more prudent solutions from experienced devs.

Reply:
The easiest fix would be to have a boolean variable, which would be set to true inside the while loop. In the end, after the loop you check if this variable is true `pop_back()` in that case.

Reply:
I think it would be best if this function just read stream content in binary mode without any line splitting. That's what `readFileAsString()` already does for files stored on disk.